packages/rum-core/src/domain/rumEventsCollection/action/trackActions.ts

```diff
@@ -136,18 +131,16 @@ class PendingAutoAction {
     this.lifeCycle.notify(LifeCycleEventType.AUTO_ACTION_CREATED, { id: this.id, startClocks: this.startClocks })
   }
 
-  complete(endClocks: ClocksState) {
-    const actionDuration = elapsed(this.startClocks.timeStamp, endClocks.timeStamp)
-    if (actionDuration < 0) {
-      addMonitoringMessage('auto action with negative loading time', {
-        debug: {
-          actionDuration,
-          type: this.type,
-          name: this.name,
-          startClocks: this.startClocks,
-          endClocks,
-        },
-      })
+  complete(event: IdlePageEvent) {
+    if (!event.hadActivity) {
+      this.discard()
+      return
+    }
+
+    const duration = elapsed(this.startClocks.timeStamp, event.endClocks.timeStamp)
+    if (duration < 0) {
+      this.discard()
+      return
```

Comment:
since `complete` and `discard` are two separate behaviors, what about keeping them separated?
we could only pass duration to `complete` if you want to avoid to compute it twice